src/panels/CharmsAndActionsPanel/CharmsAndActionsPanel.tsx

```diff
@@ -62,10 +77,32 @@ const CharmsAndActionsPanel = () => {
       .then((charmsURL) => {
         const isCharmURLUnique = charmsURL.length === 1;
         setCharmURL(isCharmURLUnique ? charmsURL[0] : null);
+        setInlineErrors(InlineErrors.GET_URL, null);
         return;
       })
-      .catch((error) => console.error(Label.GET_URL_ERROR, error));
-  }, [appState, dispatch, isPanelLoading, modelUUID, selectedApplications]);
+      .catch((error) => {
+        setInlineErrors(InlineErrors.GET_URL, Label.GET_URL_ERROR);
+        console.error(Label.GET_URL_ERROR, error);
+      });
+  }, [appState, dispatch, modelUUID, selectedApplications, setInlineErrors]);
```

Comment:
This useCallback is passed `appState` which means that this callback gets recalculated every time the redux store changes. If there's an error during the `getCharmsURLFromApplications` it means that this will enter an infinite loop while the store gets updated, this callback gets recalculated and then the useEffect below gets run (to see this add `throw new Error();` before the return in `getCharmsURLFromApplications` in api.ts).

To prevent this we can change `const appState = useAppStore().getState();` to instead be a reference to the `getState` function `const getState = useAppStore().getState;` which won't change and then call that inside the `useCallback`.

Reply:
Thanks for the explanation! Fixed.